modules/oe_content_sub_entity/modules/oe_content_sub_entity_person/src/PersonNodeUpdater.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_content_sub_entity_person;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\node\NodeInterface;
+
+/**
+ * Updates a given node's 'oe_persons' field values to sub-entities.
+ *
+ * The created sub-entities will be assigned to 'oe_persons_reference' field.
+ */
+class PersonNodeUpdater {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs a new PersonNodeUpdater.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager) {
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * Updates the revisions of a node with new person values.
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   The node.
+   */
+  public function updateNode(NodeInterface $node): void {
+    if (!$node->hasField('oe_persons')) {
+      return;
+    }
+
+    /** @var \Drupal\oe_content_sub_entity_person\Entity\PersonInterface $person */
+    $person = $this->entityTypeManager->getStorage('oe_person')->create([
+      'type' => 'oe_person',
+    ]);
+
+    $ids = $this->entityTypeManager->getStorage('node')->getQuery()
+      ->allRevisions()
+      ->condition('nid', $node->id())
+      ->sort('vid')
+      ->execute();
```

Comment:
We are using entity query here, all entity queries are executed with access check unless it is explicitly specified not to. For what we are doing here, we don't want access checks to be performed. Please use accessCheck(FALSE) in the query.